tests/set-prop-deep.test.ts

```diff
@@ -0,0 +1,58 @@
+import { setPropDeep } from '../src/utils/object-utils';
+
+describe('setPropDeep(obj, path, value)', () => {
+  it('should return value if no path provided', () => {
+    const value = { some: 'value' };
+    const result = setPropDeep({}, '', value);
+
+    expect(result).toBe(value);
+  });
+
+  it('should should return new object with the value put on the specified path if no obj provided', () => {
+    const value = { some: 'value' };
+    const result = setPropDeep(null, 'a.b.c', value);
+
+    const acceptedResult = {
+      a: {
+        b: {
+          c: value,
+        },
+      },
+    };
+
+    expect(result).toStrictEqual(acceptedResult);
+  });
+
+  it('should return obj with the value put on the specified path', () => {
+    const value = { some: 'value' };
+    const result = setPropDeep({ a: { e: 'f' }, g: { h: 'l' } }, 'a.b.c', value);
```

Comment:
have `const obj = { a: { e: 'f' }, g: { h: 'l' } }`. and then have `objClone = JSON.parse(JSON.stringify(obj))`. and change objClone instead of creating that big obj... It's just hard for the reader of the code to compare them...